react/index.d.ts

```diff
@@ -37,7 +37,7 @@ declare namespace React {
 
     type CElement<P, T extends Component<P, ComponentState>> = ComponentElement<P, T>;
     interface ComponentElement<P, T extends Component<P, ComponentState>> extends ReactElement<P> {
-        type: ComponentClass<P>;
+        type: ComponentClass<P> | ComponentClass<any>;
```

Comment:
I suppose that doesn't fix your test case of `React.cloneElement(element, {}, null)`. Is it possible to use a `null` literal type? ie.

```ts
function cloneElement<P>(element: ReactElement<P>, props: null, ...children: ReactNode[]): ReactElement<P>;
```